Now Travis is failing:

```
Running tests from 'tests/win-unicode' ...
 ... testing 'exec_tests.ml': => skipped
 ... testing 'mltest.ml': => skipped
 ... testing 'symlink_tests.ml': => skipped
Running tests from 'tests/array-functions' ...
/bin/sh: 5: Bad substitution
Running tests from 'tests/basic-io-2' ...
/bin/sh: 5: Bad substitution
Running tests from 'tests/basic-io' ...
/bin/sh: 5: Bad substitution
Running tests from 'tests/basic-modules' ...
/bin/sh: 5: Bad substitution
Running tests from 'tests/basic-multdef' ...
/bin/sh: 5: Bad substitution
Running tests from 'tests/basic-private' ...
/bin/sh: 5: Bad substitution
make[1]: *** [new-without-report] Error 1
```

I'm not sure why only the very last tests are failing, but one error I see in the current diff is that rebasing reintroduced the `\r` removal in `$(ocamltest) $$dir/$${testfile/$$'\r'}` that was removed by #1441, I think that should be just `$$dir/$$testfile` now that we have the IFS business.

--- a/.gitignore
+++ b/.gitignore
@@ -241,6 +241,7 @@ _ocamltest
 /testsuite/**/*.native
 /testsuite/**/program
 /testsuite/**/_log
+/testsuite/failure.stamp
 
 /testsuite/_retries
 

--- a/testsuite/Makefile
+++ b/testsuite/Makefile
@@ -39,6 +39,8 @@ else # Windows
   endif
 endif
 
+failstamp := failure.stamp
+
 ocamltest_directory := ../ocamltest
 
 ocamltest_program := $(or \
@@ -97,14 +99,17 @@ new:
 
 .PHONY: new-without-report
 new-without-report: lib tools
-	@for file in `$(find) tests -name ocamltests`; do \
+	@rm -f $(failstamp)
+	@(for file in `$(find) tests -name ocamltests`; do \
 	  dir=`dirname $$file`; \
 	  echo Running tests from \'$$dir\' ... ; \
-	  IFS=$$(printf "\r\n"); while read testfile; do \
+	  (IFS=$$(printf "\r\n"); while read testfile; do \
 	    TERM=dumb OCAMLRUNPARAM= \
-	      $(ocamltest) $$dir/$$testfile; \
-	  done < $$file; \
-	done 2>&1 | tee -a _log
+	      $(ocamltest) $$dir/$$testfile || \
+	      touch $(failstamp); \
+	  done < $$file) || touch $(failstamp); \
+	done || touch $(failstamp)) 2>&1 | tee -a _log
+	@if [ -f $(failstamp) ]; then rm $(failstamp); exit 1; fi
 
 .PHONY: all-%
 all-%: lib tools
@@ -236,6 +241,7 @@ clean:
 	  (cd `dirname $$file` && $(MAKE) BASEDIR=$(BASEDIR) clean); \
 	done
 	$(FIND) . -name '*_ocamltest*' | xargs rm -rf
+	rm -f $(failstamp)
 
 .PHONY: report
 report:

